sds_data_manager/lambda_code/SDSCode/upload_api.py

```diff
@@ -12,7 +12,14 @@
 logger.setLevel(logging.INFO)
 
 BUCKET_NAME = os.getenv("S3_BUCKET")
-S3_CLIENT = boto3.client("s3")
+REGION = os.getenv("REGION")
+# The default presigned url signature does not include the region information
+# within the signature and we should be hitting the actual s3 region endpoint
+# to avoid any 307 redirects. (Generally only an issue on newly created buckets
```

Comment:
Can you add to this docstring so I remember in the future?

"to avoid any 307 redirects we use s3v4 signing method. This method includes the region in the URL, so when the user uploads a file, the URL will point directly to the correct regional S3 endpoint. "

Reply:
Yep, done.